content/en/real_user_monitoring/guide/monitor-your-nextjs-app-with-rum.md

```diff
@@ -0,0 +1,40 @@
+---
+title: Monitor Your Next.js App With RUM
+kind: guide
+description: Guide for monitoring Next.js applications with RUM.
+further_reading:
+- link: '/monitors/create/types/real_user_monitoring/'
+  tag: 'Documentation'
+  text: 'Learn about RUM Monitors'
+---
+
+## Overview
+[Next.js][1] is a JavaScript framework that is used to create React.js web pages and Node.js APIs. You can integrate Next.js with RUM to monitor your frontend applications for browser-related metrics that you give insight into performance and user behavior.
+
+## Setup
+
+To set up Datadog RUM browser monitoring:
+
+1. Navigate to **[UX Monitoring > Real User Monitoring][2]**.
+2. Click the **New Application** button.
+3. Make sure JS is selected, then enter a name for your application and click **Create New RUM Application**. A `clientToken` and `applicationId` are atuomatically generated for your application.
+4. Instrument your application through [npm][3], [CDN async][4], or [CDN sync][5].
+5. Follow the in-app steps to verify your installation.
+6. Deploy the changes to your application. Once your deployment is live, Datadog collects events from user browsers.
+7. In your `.env.local` file, prefix the update the `applicationId: process.env.DATADOG_APPLICATION_ID` variable to include `NEXT_PUBLIC_` like below:
```

Comment:
I've created a Next app using : `npx create-next-app@latest`
I cannot find a `.env.local` file out-of-the-box. . Maybe we should link the relevant information: https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables and give more context about this ? Don't we need also to update the snippet that we cut&paste in 3) ? 
Also, shouldn't we do the same for `clientToken` ?

Reply:
Does the application ID need to be exposed as an env var? I am able to get RUM sessions connected to APM without it, I'm not sure if it's having any effect unless I'm missing something

Reply:
I.e. just copying the code snippet with the applicationId hardcoded seems to work for me without issue